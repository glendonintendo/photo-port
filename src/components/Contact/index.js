import React, { useState } from 'react';

function ContactForm() {
	const [formState, setFormState] = useState({ name: '', email: '', message: ''});
	const { name, email, message } = formState;
	
	const handleChange = (e) => {
		setFormState({...formState, [e.target.name]: e.target.value });
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		console.log(formState);
	};

	return (
		<section>
			<h1>Contact Me</h1>
			
			<form id='contact-form' onSubmit={handleSubmit}>
				<div>
					<label htmlFor='name'>Name:</label>
					<input 
						type ='text' 
						name='name' 
						default={name} 
						onChange={handleChange}					
					/>
				</div>

				<div>
					<label htmlFor='email'>Email Address:</label>
					<input 
						type='email' 
						name='email' 
						default={email} 
						onChange={handleChange}	
					/>
				</div>

				<div>
					<label htmlFor='message'>Message:</label>
					<textarea 
						name='message' 
						rows='5' 
						default={message} 
						onChange={handleChange}	
					/>
				</div>

				<button type='submit'>Submit</button>
			</form>
		</section>
	)
};

export default ContactForm;