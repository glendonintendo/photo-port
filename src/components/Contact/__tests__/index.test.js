import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

afterEach(cleanup);

describe('Contact form', () => {
	it('renders', () => {
		render(<Contact />);
	});

	it('matches snapshot', () => {
		const { asFragment } = render(<Contact />);

		expect(asFragment()).toMatchSnapshot();
	});
});

describe('Contact title', () => {
	it('renders', () => {
		const { getByTestId } = render(<Contact />);
		
		expect(getByTestId('form-title')).toHaveTextContent('Contact Me');
		expect(getByTestId('form-button')).toHaveTextContent('Submit');
	});
});