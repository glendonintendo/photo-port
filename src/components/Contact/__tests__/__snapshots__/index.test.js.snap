// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact form matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="form-title"
    >
      Contact Me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email Address:
        </label>
        <input
          name="email"
          type="email"
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="form-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
